clang/test/dpct/helper_files_ref/include/dpl_extras/algorithm.h

```diff
@@ -1208,6 +1349,48 @@ inline void sort_keys(
   }
 }
 
+template <typename _ExecutionPolicy, typename key_t, typename value_t,
+          typename OffsetIteratorT>
+inline void segmented_sort_pairs(
+    _ExecutionPolicy &&policy, key_t keys_in, key_t keys_out, value_t values_in,
+    value_t values_out, int64_t n, int64_t nsegments,
+    OffsetIteratorT begin_offsets, OffsetIteratorT end_offsets,
+    bool descending = false, int begin_bit = 0,
+    int end_bit = sizeof(typename ::std::iterator_traits<key_t>::value_type) *
+                  8) {
+  int compute_units =
+      policy.queue()
+          .get_device()
+          .template get_info<sycl::info::device::max_compute_units>();
+  auto sg_sizes = policy.queue()
+                      .get_device()
+                      .template get_info<sycl::info::device::sub_group_sizes>();
+  int subgroup_size = sg_sizes.empty() ? 1 : sg_sizes.back();
+  int compute_unit_threads =
+      policy.queue().get_device().is_gpu() ? subgroup_size : 1;
```

Comment:
subgroup size is width of SMID, while threads of compute unit is 7 or 8 (fixed value),  so the logic there is to cal: simd or compute unit thread? Maybe need update the variable name.

Reply:
Adjusted the code to remove this variable name to prevent confusion.